app/process_proposal.go

```diff
@@ -118,73 +126,78 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 		// - that the sizes match
 		// - that the namespaces match between blob and PFB
 		// - that the share commitment is correct
-		if err := blobtypes.ValidateBlobTx(app.txConfig, blobTx, subtreeRootThreshold, app.AppVersion()); err != nil {
-			logInvalidPropBlockError(app.Logger(), req.Header, fmt.Sprintf("invalid blob tx %d", idx), err)
-			return reject()
+
+		v, err := app.AppVersion(app.NewContext(false))
+		if err != nil {
+			logInvalidPropBlockError(app.Logger(), blockHeader, "failure to get app version", err)
+		}
+
+		if err := blobtypes.ValidateBlobTx(app.encodingConfig.TxConfig, blobTx, subtreeRootThreshold, v); err != nil {
+			logInvalidPropBlockError(app.Logger(), blockHeader, fmt.Sprintf("invalid blob tx %d", idx), err)
+			return reject(), nil
```

Comment:
[question]

can we use the app version from above?

[note]
we might want to keep the app version as an arg to `blobtypes.ValidateBlobTx` to enable downstream repos to only have to import a single version of the app